web-api/src/storage/postgres.rs

```diff
@@ -284,89 +297,154 @@ impl storage::Interest for Storage {
         Ok(UserInterests { positive, negative })
     }
 
-    async fn update_positive<F>(
-        &self,
-        doc_id: &DocumentId,
+    async fn upsert_cois(
+        tx: &mut Transaction<'_, Postgres>,
+        user_id: &UserId,
+        now: DateTime<Utc>,
+        cois: &[PositiveCoi],
+    ) -> Result<(), Error> {
+        if cois.is_empty() {
+            return Ok(());
+        }
+        //FIXME micro benchmark and chunking+persist abstraction
+        let persist = cois.len() < 10;
```

Comment:
We have been doing the "micro optimization" of not persisting potentially widely varying queries since before the last pivot. I just slightly improved this  existing practice by persisting the query for the most common case of one value. I'm a bit stumped that  after  changing an untested optimization to a obvious better but still as much untested optimization caused us to have to discuss this now in this PR.

 > The maximum number of raw that we can insert is bound from the API to 100
 
 yes, but that's mean implicitly relying on a current state of business logic in a way which will most likely not be caught/found if that logic ever changes. Sure it's not likely to change, but a bad idea anyway.